### DISCUSSION TITLE
Is it possible to map roles under groups in token?

### DISCUSSION THREAD

**Discussion by late-e** [keycloak/keycloak#26429] [Q&A]
I have multiple organizations in my application, and one user might have permission for multiple organization. Organization is a group, and there is subgroups like "admins" and "basic users". Client roles are mapped to those subgroups.
Now when I set user to multiple organizations and their subgroups, I have no way of knowing, that what group gives what roles to the user.
So Organization 1 could give "admin" role for the user, and Organization 2 gives "basic" role for the user. But in Token, I have no way to know that where those roles come from. 
So is there a way to map roles in token under the groups, so I would know that what role is for what organization?

Thank you for your help.

**Comment by late-e** [comment]
This need was solved with custom protocol mapper.

**Comment by Remsboys** [comment]
@late-e Hi, I have the same question. Can you send me the configuration/code of your mapper? Thank you very much!

**Comment by T-Herrmann-WI** [comment]
Can you send me the configuration of your mapper too? A screenshot would be enough. Thank you !

**Reply by Remsboys** [reply]
Hi @T-Herrmann-WI,
In the meantime, I implemented a mapper myself. I uploaded the code to Gist for you: https://gist.github.com/Remsboys/d4e62c1d5daff0a4459ca846e9f85d1a

**Reply by late-e** [reply]
Hi,
looks good, mine is very similar. Unfortunately I cannot share it as it's company code.
